### PR TITLE
The Bricks Documentation Project.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

**Comment by dkorolev** [comment]
And we have `cerealize`, `rtti` and `net/api` covered with documentation.

(Will require some polishing, but I'd like to get Penny-style arithmetics to John before going to bed.)

@sompylasar @deathbaba PTAL.

Thanks,
Dima

**Comment by sompylasar** [comment]
"use CEREAL_NVP(property)" -> "CEREAL_NVP(property) or cereal::make_nvp("property", name_we_want_to_change)" if this doesn't make things complex (the latter is not easy to find in the cereal docs)

File: README.md [hunk 1]
```diff
@@ -30,11 +41,240 @@ x.vector_int.push_back(2);
...
+// `JSONParse<T>(json)` creates an instance of T from a JSON.
 const SimpleType y = JSONParse<SimpleType>(json);
-// `JSONParse(json, T&)` is an alternate two-parameters form that enables omitting the template type.
+
+// `JSONParse(json, T& out)` allows omitting the type.
 SimpleType z;
 JSONParse(json, z);
 ```
+```cpp
+// Use `load()/save()` instead of `serialize()` to customize serialization.
```
Comment (deathbaba) [inline README.md]: Why duplicate cereal's docs? Just put a link there should be enough, no?

**Comment by dkorolev** [comment]
@sompylasar , @deathbaba :

1. In our case, we just serialize fields, and I'd prefer to keep it this way, Thus, ```cereal::make_nvp``` is a bit too advanced of a usecase.

2. No, it's not enough to put a link to Cereal's docs. First, any documentation that is short, self- contained and has examples, is better than a link somewhere outside. Second, JSON-wise, our usecase of Cereal is rather non-straightforward, with a or of Bricks-specific code, and it's best to keep the documentation in one place.  Lastly, the link is already there :)

**Comment by dkorolev** [comment]
Thanks!

On Thursday, February 19, 2015, Alex Zolotarev <notifications@github.com>
wrote:

> Merged #68 <https://github.com/KnowSheet/Bricks/pull/68>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KnowSheet/Bricks/pull/68#event-238681310>.
>


-- 
Thanks,
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783

**Pull request merged**